# SFF2FASTA
Python script to convert SFF to FASTA.

# DOWNLOADING THE TOOL
```
git clone --recursive https://github.com/shuko-tech/SFF2FASTA.git
cd ./SFF2FASTA
```

# RUNNING THE TOOL
```
usage: ssf2fasta.py [-h] [-i INPUT_SFF] [-o OUTPUT_FASTA] [-b BATCH_SIZE]

		SFF to FASTA Conversion Tool

		optional arguments:
		-h, --help            show this help message and exit
		-i INPUT_SFF, --input_sff INPUT_SFF
								Full path to input .sff file or directory containing .sff files. Required.
		-o OUTPUT_FASTA, --output_fasta OUTPUT_FASTA
								Directory to output .fasta file or files. Default: Same directory as input. Optional.
		-b BATCH_SIZE, --batch_size BATCH_SIZE
								Batch size. Default: No batching.
		-V VERBOSITY, --verbosity VERBOSITY
								Verbosity mode for output. [0, Log Only; 1, Log and Print to Console; 2, Print Only].
```

# SFF2FASTQ SUBMODULE

Indraniel Das (indraniel@gmail.com or idas@wustl.edu) The Genome Institute at Washington University

This software was developed at The Genome Institute at Washington University, St. Louis, MO.

## DISCLAIMER

The ssf2fastq software is not owned by shuko-tech. 

See auhors project repository here: [https://github.com/indraniel/sff2fastq](https://github.com/indraniel/sff2fastq)

This software is provided "as is" without warranty of any kind.


# SFF2FASTQ SUBMODULE

## DISCLAIMER

The fastq2fasta software is not owned by shuko-tech. 

See auhors project repository here: [https://github.com/dantaki/fastq2fasta](https://github.com/dantaki/fastq2fasta)

This software is provided "as is" without warranty of any kind.

----
----

##### April 09, 2021